crates/router/src/configs/settings.rs

```diff
@@ -676,7 +676,13 @@ impl Settings<SecuredSecret> {
                     .with_list_parse_key("redis.cluster_urls")
                     .with_list_parse_key("events.kafka.brokers")
                     .with_list_parse_key("connectors.supported.wallets")
-                    .with_list_parse_key("connector_request_reference_id_config.merchant_ids_send_payment_id_as_connector_request_id"),
+                    .with_list_parse_key("connector_request_reference_id_config.merchant_ids_send_payment_id_as_connector_request_id")
```

Comment:
This is not scalable: this would not allow us to use any suitable key (any possible enum variant) in the `HashMap`, but only restricts to the specified keys. We would need to address this in a separate PR.